File: app/src/main/java/com/punchlist/app/data/model/Status.kt
package com.punchlist.app.data.model

import androidx.compose.ui.graphics.Color

enum class Status(val label: String, val color: Color) {
    OPEN("Open", Color(0xFF2196F3)),
    IN_PROGRESS("In Progress", Color(0xFFFF9800)),
    NEEDS_REVIEW("Needs Review", Color(0xFF9C27B0)),
    COMPLETE("Complete", Color(0xFF4CAF50));

    companion object {
        fun fromString(value: String): Status =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) } ?: OPEN
    }
}

File: PunchlistApp/Models/Status.cs
using Microsoft.Maui.Graphics;

namespace PunchlistApp.Models;

public enum Status
{
    Open,
    InProgress,
    NeedsReview,
    Complete
}

public static class StatusExtensions
{
    public static string Label(this Status s) => s switch
    {
        Status.Open => "Open",
        Status.InProgress => "In Progress",
        Status.NeedsReview => "Needs Review",
        Status.Complete => "Complete",
        _ => "Unknown"
    };

    public static Color Color(this Status s) => s switch
    {
        Status.Open => Microsoft.Maui.Graphics.Color.FromArgb("#2196F3"),
        Status.InProgress => Microsoft.Maui.Graphics.Color.FromArgb("#FF9800"),
        Status.NeedsReview => Microsoft.Maui.Graphics.Color.FromArgb("#9C27B0"),
        Status.Complete => Microsoft.Maui.Graphics.Color.FromArgb("#4CAF50"),
        _ => Microsoft.Maui.Graphics.Color.FromArgb("#9E9E9E")
    };
}
